jdbc/src/java/org/apache/hive/jdbc/HiveConnection.java

```diff
@@ -1324,6 +1329,13 @@ private boolean isKerberosAuthMode() {
         && sessConfMap.containsKey(JdbcConnectionParams.AUTH_PRINCIPAL);
   }
 
+  private boolean isEnableCanonicalHostnameCheck() {
+    return Optional.ofNullable(sessConfMap.get(JdbcConnectionParams.AUTH_KERBEROS_ENABLE_CANONICAL_HOSTNAME_CHECK))
```

Comment:
I think this can be simplified with Boolean.parseBoolean(sessConfMap.getOrDefault(..., "true"))

Reply:
true my bad; fixed :)